File: docs-examples/example-java/src/test/java/io/pollend/quartz/docs/quickstart/MessageScheduler.java
package io.pollend.quartz.docs.quickstart;

// tag::imports[]
import io.pollend.quartz.annotation.QuartzKey;
import io.pollend.quartz.annotation.QuartzSchedule;
import io.pollend.quartz.annotation.ScheduleOn;
// end:imports[]

// tag::clazz[]
@QuartzSchedule()
public interface MessageScheduler {
    @ScheduleOn(MessageJob.class)
    void sendMessage(@QuartzKey("message") String message);
}
// end::clazz[]

File: docs-examples/example-kotlin/src/test/kotlin/io/pollend/quartz/docs/quickstart/MessageScheduler.kt
package io.pollend.quartz.docs.quickstart

// tag::imports[]
import io.pollend.quartz.annotation.QuartzKey
import io.pollend.quartz.annotation.QuartzSchedule
import io.pollend.quartz.annotation.ScheduleOn
import org.quartz.JobKey
// end:imports[]

// tag::clazz[]
@QuartzSchedule()
interface MessageScheduler {
    @ScheduleOn(MessageJob::class)
    fun sendMessage(key: JobKey, @QuartzKey("message") message: String)
}
// end::clazz[]
